### ISSUE TITLE
Inconsistent results in object detection with supervised backbone

### ISSUE THREAD

**Issue by liming-ai** [open-mmlab/mmselfsup#558]
### Branch

master branch (0.x version, such as `v0.10.0`, or `dev` branch)

### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmselfsup/issues) and [Discussions](https://github.com/open-mmlab/mmselfsup/discussions) but cannot get the expected help.
- [X] I have read the [documentation](https://mmselfsup.readthedocs.io/en/latest/) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmselfsup).

### Environment

Any

### Describe the bug

When I use [mmdet config](https://github.com/open-mmlab/mmselfsup/blob/master/configs/benchmarks/mmdetection/coco/mask_rcnn_r50_fpn_mstrain_1x_coco.py) to train the Mask R-CNN model, its box mAP is 40.1, which is much higher than 38.9 of [detectron2 config](https://github.com/open-mmlab/mmselfsup/blob/master/configs/benchmarks/detectron2/coco_R_50_FPN_1x_moco.yaml). The default torchvision R50 supervised backbone is used.

Are there any inconsistent between  [mmdet config](https://github.com/open-mmlab/mmselfsup/blob/master/configs/benchmarks/mmdetection/coco/mask_rcnn_r50_fpn_mstrain_1x_coco.py) and [detectron2 config](https://github.com/open-mmlab/mmselfsup/blob/master/configs/benchmarks/detectron2/coco_R_50_FPN_1x_moco.yaml)?

**Comment by fangyixiao18** [comment]
Hi, we checked the config. Maybe you could change [Shared4Conv1FCBBoxHead](https://github.com/open-mmlab/mmselfsup/blob/master/configs/benchmarks/mmdetection/coco/mask_rcnn_r50_fpn_mstrain_1x_coco.py#L12) to default [Shared2FCBBoxHead](https://github.com/open-mmlab/mmselfsup/blob/master/configs/benchmarks/mmdetection/_base_/models/mask_rcnn_r50_fpn.py#L43) and try it again.

**Comment by liming-ai** [comment]
> Hi, we checked the config. Maybe you could change [Shared4Conv1FCBBoxHead](https://github.com/open-mmlab/mmselfsup/blob/master/configs/benchmarks/mmdetection/coco/mask_rcnn_r50_fpn_mstrain_1x_coco.py#L12) to default [Shared2FCBBoxHead](https://github.com/open-mmlab/mmselfsup/blob/master/configs/benchmarks/mmdetection/_base_/models/mask_rcnn_r50_fpn.py#L43) and try it again.

Hi, I understand that there are differences between `4conv1fc` and `2fc`. However, as reported in MoCo, the supervised pre-trained Mask R-CNN with `4conv1fc` should be 38.9 mAP, and I can reproduce the result. However, the result from mmdet config is much higher than 38.9, that's my concern

**Comment by fangyixiao18** [comment]
> Hi, I understand that there are differences between `4conv1fc` and `2fc`. However, as reported in MoCo, the supervised pre-trained Mask R-CNN with `4conv1fc` should be 38.9 mAP, and I can reproduce the result. However, the result from mmdet config is much higher than 38.9, that's my concern

It is recommend that you create an issue in MMDet to get more help. Thanks
